### PR TITLE
Issues 322 and 324

### PR DESCRIPTION
**Pull request by dominik-mokris-mtu**
This pull request solves issues #322 (by improving the documentation of `active_into(...)`) and #324 (by fixing a bug in `gsKnotVector`).

# The checklist:

> - [X] Have you added an explanation of what your changes do and why
  you'd like us to include them?

Yes, this has been explained in issues #322 and #324.

> - [ ] Have you documented any new codes using Doxygen comments?

No, since it is a bugfix and an improvement of the documentation.

> - [ ] Have you written new tests or examples for your changes?

No, but I ran the unittests. Do you think I should write a new unittest for the change in `gsKnotVector`?

### CODE REVIEW

**Comment by dominik-mokris-mtu** [comment]
I was too fast and I did not notice a bug I introduced. Once it is solved, I will either re-open the pull request or create a new one.

**Comment by dominik-mokris-mtu** [comment]
The new bug is corrected. Funny enough, it had worked on Linux but not on Mac and Windows.

File: src/gsNurbs/gsKnotVector.hpp [hunk 1]
```diff
@@ -635,12 +635,14 @@ void gsKnotVector<T>::initUniform( T first,
     for(unsigned i = m_deg - mult_ends + 1, j=0; i!= 0; --i)
```
Comment (codewerfer) [inline src/gsNurbs/gsKnotVector.hpp]: you can also write `for(unsigned i = m_deg - mult_ends + 1, j=0; i!= 0; --i, ++j)`

File: src/gsNurbs/gsKnotVector.hpp [hunk 2]
```diff
@@ -635,12 +635,14 @@ void gsKnotVector<T>::initUniform( T first,
...
     {   // add left ghost knots
         m_repKnots.push_back(first-i*h);
-	++j;
-        m_multSum .push_back(j);
+        m_multSum .push_back(++j);
     }
 
     m_repKnots.insert(m_repKnots.end(), mult_ends, first);
-    m_multSum .push_back(m_multSum.back() + mult_ends);
+    if(m_multSum.size() == 0)
```
Comment (codewerfer) [inline src/gsNurbs/gsKnotVector.hpp]: `m_multSum.push_back(mult_ends + (m_multSum.size() == 0 ? 0 : m_multSum.back()));` is a quite better code to optimize.

<details><summary>See: (click on me)</summary>
Left: yours

Right: mine
<p>

```
td::vector<int, std::allocator<int> >::~vector() [base objec   std::vector<int, std::allocator<int> >::~vector() [base objec
        mov     rdi, QWORD PTR [rdi]                                    mov     rdi, QWORD PTR [rdi]
        test    rdi, rdi                                                test    rdi, rdi
        je      .L1                                                     je      .L1
        jmp     operator delete(void*)                                  jmp     operator delete(void*)
.L1:                                                            .L1:
        ret                                                             ret
.LC0:                                                           .LC0:
        .string "vector::_M_realloc_insert"                             .string "vector::_M_realloc_insert"
void std::vector<int, std::allocator<int> >::_M_realloc_inser | void std::vector<int, std::allocator<int> >::_M_realloc_inser
        movabs  rcx, 2305843009213693951                                movabs  rcx, 2305843009213693951
        push    r15                                                     push    r15
        push    r14                                                     push    r14
        push    r13                                                     push    r13
        push    r12                                                     push    r12
        push    rbp                                                     push    rbp
        push    rbx                                                     push    rbx
        sub     rsp, 24                                                 sub     rsp, 24
        mov     r12, QWORD PTR [rdi+8]                                  mov     r12, QWORD PTR [rdi+8]
        mov     r8, QWORD PTR [rdi]                                     mov     r8, QWORD PTR [rdi]
        mov     rax, r12                                                mov     rax, r12
        sub     rax, r8                                                 sub     rax, r8
        sar     rax, 2                                                  sar     rax, 2
        cmp     rax, rcx                                                cmp     rax, rcx
        je      .L18                                                    je      .L18
        mov     r15, rdx                                                mov     r15, rdx
        mov     rdx, rsi                                                mov     rdx, rsi
        mov     rbp, rdi                                                mov     rbp, rdi
        mov     r13, rsi                                                mov     r13, rsi
        sub     rdx, r8                                                 sub     rdx, r8
        test    rax, rax                                                test    rax, rax
        je      .L14                                                    je      .L14
        movabs  r14, 9223372036854775804                                movabs  r14, 9223372036854775804
        lea     rsi, [rax+rax]                                          lea     rsi, [rax+rax]
        cmp     rax, rsi                                                cmp     rax, rsi
        jbe     .L19                                                    jbe     .L19
.L6:                                                            .L6:
        mov     rdi, r14                                                mov     rdi, r14
        mov     QWORD PTR [rsp+8], rdx                                  mov     QWORD PTR [rsp+8], rdx
        mov     QWORD PTR [rsp], r8                                     mov     QWORD PTR [rsp], r8
        call    operator new(unsigned long)                             call    operator new(unsigned long)
        mov     r8, QWORD PTR [rsp]                                     mov     r8, QWORD PTR [rsp]
        mov     rdx, QWORD PTR [rsp+8]                                  mov     rdx, QWORD PTR [rsp+8]
        mov     rbx, rax                                                mov     rbx, rax
        add     r14, rax                                                add     r14, rax
.L13:                                                           .L13:
        mov     eax, DWORD PTR [r15]                                    mov     eax, DWORD PTR [r15]
        sub     r12, r13                                                sub     r12, r13
        lea     r15, [rbx+4+rdx]                                        lea     r15, [rbx+4+rdx]
        mov     DWORD PTR [rbx+rdx], eax                                mov     DWORD PTR [rbx+rdx], eax
        lea     rax, [r15+r12]                                          lea     rax, [r15+r12]
        mov     QWORD PTR [rsp], rax                                    mov     QWORD PTR [rsp], rax
        test    rdx, rdx                                                test    rdx, rdx
        jg      .L20                                                    jg      .L20
        test    r12, r12                                                test    r12, r12
        jg      .L9                                                     jg      .L9
        test    r8, r8                                                  test    r8, r8
        jne     .L12                                                    jne     .L12
.L10:                                                           .L10:
        movq    xmm0, rbx                                               movq    xmm0, rbx
        mov     QWORD PTR [rbp+16], r14                                 mov     QWORD PTR [rbp+16], r14
        movhps  xmm0, QWORD PTR [rsp]                                   movhps  xmm0, QWORD PTR [rsp]
        movups  XMMWORD PTR [rbp+0], xmm0                               movups  XMMWORD PTR [rbp+0], xmm0
        add     rsp, 24                                                 add     rsp, 24
        pop     rbx                                                     pop     rbx
        pop     rbp                                                     pop     rbp
        pop     r12                                                     pop     r12
        pop     r13                                                     pop     r13
        pop     r14                                                     pop     r14
        pop     r15                                                     pop     r15
        ret                                                             ret
.L20:                                                           .L20:
        mov     rsi, r8                                                 mov     rsi, r8
        mov     rdi, rbx                                                mov     rdi, rbx
        mov     QWORD PTR [rsp+8], r8                                   mov     QWORD PTR [rsp+8], r8
        call    memmove                                                 call    memmove
        test    r12, r12                                                test    r12, r12
        mov     r8, QWORD PTR [rsp+8]                                   mov     r8, QWORD PTR [rsp+8]
        jg      .L9                                                     jg      .L9
.L12:                                                           .L12:
        mov     rdi, r8                                                 mov     rdi, r8
        call    operator delete(void*)                                  call    operator delete(void*)
        jmp     .L10                                                    jmp     .L10
.L19:                                                           .L19:
        test    rsi, rsi                                                test    rsi, rsi
        jne     .L7                                                     jne     .L7
        xor     r14d, r14d                                              xor     r14d, r14d
        xor     ebx, ebx                                                xor     ebx, ebx
        jmp     .L13                                                    jmp     .L13
.L9:                                                            .L9:
        mov     rdx, r12                                                mov     rdx, r12
        mov     rsi, r13                                                mov     rsi, r13
        mov     rdi, r15                                                mov     rdi, r15
        mov     QWORD PTR [rsp+8], r8                                   mov     QWORD PTR [rsp+8], r8
        call    memcpy                                                  call    memcpy
        mov     r8, QWORD PTR [rsp+8]                                   mov     r8, QWORD PTR [rsp+8]
        test    r8, r8                                                  test    r8, r8
        je      .L10                                                    je      .L10
        jmp     .L12                                                    jmp     .L12
.L14:                                                           .L14:
        mov     r14d, 4                                                 mov     r14d, 4
        jmp     .L6                                                     jmp     .L6
.L18:                                                           .L18:
        mov     edi, OFFSET FLAT:.LC0                                   mov     edi, OFFSET FLAT:.LC0
        call    std::__throw_length_error(char const*)                  call    std::__throw_length_error(char const*)
.L7:                                                            .L7:
        cmp     rsi, rcx                                                cmp     rsi, rcx
        cmovbe  rcx, rsi                                                cmovbe  rcx, rsi
        mov     r14, rcx                                                mov     r14, rcx
        sal     r14, 2                                                  sal     r14, 2
        jmp     .L6                                                     jmp     .L6
void std::vector<int, std::allocator<int> >::_M_realloc_inser | g(int):
        movabs  rcx, 2305843009213693951                      <
        push    r15                                           <
        push    r14                                           <
        push    r13                                           <
        push    r12                                           <
        push    rbp                                           <
        push    rbx                                           <
        sub     rsp, 24                                                 sub     rsp, 24
        mov     r12, QWORD PTR [rdi+8]                        |         mov     rsi, QWORD PTR vec[rip+8]
        mov     r8, QWORD PTR [rdi]                           |         cmp     QWORD PTR vec[rip], rsi
        mov     rax, r12                                      |         je      .L22
        sub     rax, r8                                       |         add     edi, DWORD PTR [rsi-4]
        sar     rax, 2                                        | .L22:
        cmp     rax, rcx                                      |         cmp     rsi, QWORD PTR vec[rip+16]
        je      .L35                                          <
        mov     r15, rdx                                      <
        mov     rdx, rsi                                      <
        mov     rbp, rdi                                      <
        mov     r13, rsi                                      <
        sub     rdx, r8                                       <
        test    rax, rax                                      <
        je      .L31                                          <
        movabs  r14, 9223372036854775804                      <
        lea     rsi, [rax+rax]                                <
        cmp     rax, rsi                                      <
        jbe     .L36                                          <
.L23:                                                         <
        mov     rdi, r14                                      <
        mov     QWORD PTR [rsp+8], rdx                        <
        mov     QWORD PTR [rsp], r8                           <
        call    operator new(unsigned long)                   <
        mov     r8, QWORD PTR [rsp]                           <
        mov     rdx, QWORD PTR [rsp+8]                        <
        mov     rbx, rax                                      <
        add     r14, rax                                      <
.L30:                                                         <
        mov     eax, DWORD PTR [r15]                          <
        sub     r12, r13                                      <
        lea     r15, [rbx+4+rdx]                              <
        mov     DWORD PTR [rbx+rdx], eax                      <
        lea     rax, [r15+r12]                                <
        mov     QWORD PTR [rsp], rax                          <
        test    rdx, rdx                                      <
        jg      .L37                                          <
        test    r12, r12                                      <
        jg      .L26                                          <
        test    r8, r8                                        <
        jne     .L29                                          <
.L27:                                                         <
        movq    xmm0, rbx                                     <
        mov     QWORD PTR [rbp+16], r14                       <
        movhps  xmm0, QWORD PTR [rsp]                         <
        movups  XMMWORD PTR [rbp+0], xmm0                     <
        add     rsp, 24                                       <
        pop     rbx                                           <
        pop     rbp                                           <
        pop     r12                                           <
        pop     r13                                           <
        pop     r14                                           <
        pop     r15                                           <
        ret                                                   <
.L37:                                                         <
        mov     rsi, r8                                       <
        mov     rdi, rbx                                      <
        mov     QWORD PTR [rsp+8], r8                         <
        call    memmove                                       <
        test    r12, r12                                      <
        mov     r8, QWORD PTR [rsp+8]                         <
        jg      .L26                                          <
.L29:                                                         <
        mov     rdi, r8                                       <
        call    operator delete(void*)                        <
        jmp     .L27                                          <
.L36:                                                         <
        test    rsi, rsi                                      <
        jne     .L24                                          <
        xor     r14d, r14d                                    <
        xor     ebx, ebx                                      <
        jmp     .L30                                          <
.L26:                                                         <
        mov     rdx, r12                                      <
        mov     rsi, r13                                      <
        mov     rdi, r15                                      <
        mov     QWORD PTR [rsp+8], r8                         <
        call    memcpy                                        <
        mov     r8, QWORD PTR [rsp+8]                         <
        test    r8, r8                                        <
        je      .L27                                          <
        jmp     .L29                                          <
.L31:                                                         <
        mov     r14d, 4                                       <
        jmp     .L23                                          <
.L35:                                                         <
        mov     edi, OFFSET FLAT:.LC0                         <
        call    std::__throw_length_error(char const*)        <
.L24:                                                         <
        cmp     rsi, rcx                                      <
        cmovbe  rcx, rsi                                      <
        mov     r14, rcx                                      <
        sal     r14, 2                                        <
        jmp     .L23                                          <
f(int):                                                       <
        sub     rsp, 40                                       <
        mov     rax, QWORD PTR vec[rip+8]                     <
        cmp     QWORD PTR vec[rip], rax                       <
        mov     DWORD PTR [rsp+12], edi                                 mov     DWORD PTR [rsp+12], edi
        mov     rsi, QWORD PTR vec[rip+16]                    |         je      .L23
        je      .L45                                          |         mov     DWORD PTR [rsi], edi
        mov     edx, DWORD PTR [rax-4]                        |         add     rsi, 4
        add     edx, edi                                      |         mov     QWORD PTR vec[rip+8], rsi
        mov     DWORD PTR [rsp+28], edx                       |         add     rsp, 24
        cmp     rsi, rax                                      <
        je      .L42                                          <
        mov     DWORD PTR [rax], edx                          <
        add     rax, 4                                        <
        mov     QWORD PTR vec[rip+8], rax                     <
.L38:                                                         <
        add     rsp, 40                                       <
        ret                                                   <
.L45:                                                         <
        cmp     rsi, rax                                      <
        je      .L40                                          <
        mov     DWORD PTR [rax], edi                          <
        add     rax, 4                                        <
        mov     QWORD PTR vec[rip+8], rax                     <
        add     rsp, 40                                       <
        ret                                                             ret
.L42:                                                         | .L23:
        lea     rdx, [rsp+28]                                 |         lea     rdx, [rsp+12]
        mov     edi, OFFSET FLAT:vec                                    mov     edi, OFFSET FLAT:vec
        call    void std::vector<int, std::allocator<int> >::           call    void std::vector<int, std::allocator<int> >::
        add     rsp, 40                                       |         add     rsp, 24
        ret                                                             ret
.L40:                                                         <
        lea     rdx, [rsp+12]                                 <
        mov     edi, OFFSET FLAT:vec                          <
        call    void std::vector<int, std::allocator<int> >:: <
        jmp     .L38                                          <
_GLOBAL__sub_I_vec:                                             _GLOBAL__sub_I_vec:
        pxor    xmm0, xmm0                                              pxor    xmm0, xmm0
        mov     edx, OFFSET FLAT:__dso_handle                           mov     edx, OFFSET FLAT:__dso_handle
        mov     esi, OFFSET FLAT:vec                                    mov     esi, OFFSET FLAT:vec
        mov     edi, OFFSET FLAT:_ZNSt6vectorIiSaIiEED1Ev               mov     edi, OFFSET FLAT:_ZNSt6vectorIiSaIiEED1Ev
        mov     QWORD PTR vec[rip+16], 0                                mov     QWORD PTR vec[rip+16], 0
        movaps  XMMWORD PTR vec[rip], xmm0                              movaps  XMMWORD PTR vec[rip], xmm0
        jmp     __cxa_atexit                                            jmp     __cxa_atexit
vec:                                                            vec:
        .zero   24                                                      .zero   24
```

</p>
</details>

File: src/gsNurbs/gsKnotVector.hpp (hunk 1 above)
Comment (dominik-mokris-mtu) [reply src/gsNurbs/gsKnotVector.hpp]: I like it. Incorporated in commit bbe2e008b686ef578b11efda1a53ed4dad7a1012.

File: src/gsNurbs/gsKnotVector.hpp (hunk 2 above)
Comment (dominik-mokris-mtu) [reply src/gsNurbs/gsKnotVector.hpp]: I like it. Incorporated in commit bbe2e00.

**Pull request merged**